### PR TITLE
Feature/specialized repositories

### PR DESCRIPTION
**Pull request by marcinczeczko**

### CODE REVIEW

File: documentation/src/main/wiki/ProductionUsage.md [hunk 1]
```diff
@@ -14,82 +14,93 @@ This will run the server with production settings. For more information see the
...
-          "type": "remote",
-          "path": "/content/.*",
-          "domain": "localhost",
-          "port": 3001,
+    "com.cognifide.knotx.repository.HttpRepositoryVerticle": {
+      "config": {
+        "address": "knotx.core.repository.http",
+        "configuration": {
           "client.options": {
-            "tryUseCompression": true,
```
Comment (tomaszmichalak) [inline documentation/src/main/wiki/ProductionUsage.md]: Why this line was removed?

File: knotx-core/knotx-common/src/main/java/com/cognifide/knotx/dataobjects/HttpRequestWrapper.java [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    if (formAttributes != null && !formAttributes.isEmpty()) {
+      json.put("formAttributes", toJsonArray(formAttributes));
+    }
+    return json;
+  }
+
+  private JsonArray toJsonArray(MultiMap multiMap) {
+    return multiMap.names().stream()
+        .map(name -> new JsonObject().put(name, multiMap.get(name)))
+        .reduce(new JsonArray(), (objects, item) -> objects.add(item),
```
Comment (tomaszmichalak) [inline knotx-core/knotx-common/src/main/java/com/cognifide/knotx/dataobjects/HttpRequestWrapper.java]: (objects, item) -> objects.add(item)
->
JsonArray::add

(u, u2) -> u.addAll(u2)
->
JsonArray::addAll

File: documentation/src/main/wiki/ProductionUsage.md (hunk 1 above)
Comment (marcinczeczko) [inline documentation/src/main/wiki/ProductionUsage.md]: By accident :) Fixed.

File: knotx-core/knotx-common/src/main/java/com/cognifide/knotx/dataobjects/HttpRequestWrapper.java [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+            (u, u2) -> u.addAll(u2));
+  }
+
+  private MultiMap fromJsonArray(JsonArray jsonArray) {
+    MultiMap result = MultiMap.caseInsensitiveMultiMap();
+    jsonArray.stream()
+        .map(item -> (JsonObject) item)
+        .flatMap(item -> item.stream())
```
Comment (tomaszmichalak) [inline knotx-core/knotx-common/src/main/java/com/cognifide/knotx/dataobjects/HttpRequestWrapper.java]: item -> item.stream()
->
JsonObject::stream
?

File: knotx-core/knotx-common/src/main/java/com/cognifide/knotx/dataobjects/HttpRequestWrapper.java (hunk 2 above)
Comment (marcinczeczko) [inline knotx-core/knotx-common/src/main/java/com/cognifide/knotx/dataobjects/HttpRequestWrapper.java]: fixed

File: knotx-core/knotx-common/src/main/java/com/cognifide/knotx/dataobjects/HttpRequestWrapper.java (hunk 3 above)
Comment (marcinczeczko) [inline knotx-core/knotx-common/src/main/java/com/cognifide/knotx/dataobjects/HttpRequestWrapper.java]: fixed

**Comment by tomaszmichalak** [comment]
Review done.

**Pull request merged**